xchange-stream-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingService.java

```diff
@@ -35,6 +41,11 @@
   private final boolean isPrivate;
 
   private final Map<Integer, String> subscriptionRequestMap = new ConcurrentHashMap<>();
+  private final Map<Integer, ObservableEmitter<KrakenAddOrderStatusMessage>>
+          addOrderStatusEmitters = new HashMap<>();
+
+  private final Map<Integer, ObservableEmitter<KrakenCancelOrderStatusMessage>>
+          cancelOrderStatusEmitters = new HashMap<>();
```

Comment:
just noticed this, I believe the NettyEventLoop is run with multiple (2) threads which would imply this map should be threadsafe I think these should be concurrent hashmaps.

Reply:
Fixed